nototools/ttc_utils.py

```diff
@@ -160,51 +80,58 @@ def ttc_filenames(ttc, data):
   header.
   """
     names = []
-    for font_entry in ttc.fonts:
-        name_entry = None
-        file_name = None
-        for ix in font_entry.tables:
-            if ttc.tables[ix].tag == "name":
-                name_entry = ttc.tables[ix]
-                break
-        if name_entry:
-            offset = name_entry.offset
-            limit = offset + name_entry.length
-            name_table = NameTable()
-            name_table.decompile(data[offset:limit], None)
-            ps_name = None
-            for r in name_table.names:
-                if (r.nameID, r.platformID, r.platEncID, r.langID) == (6, 3, 1, 0x409):
-                    ps_name = r.string.decode("UTF-16BE")
-                    break
-            if ps_name:
-                file_name = ps_name
-                if "-" not in ps_name:
-                    file_name += "-Regular"
-                file_name += "." + font_entry.fmt
-        names.append(file_name or ("<unknown %s>" % font_entry.fmt))
+    for font in ttc.fonts:
+        file_name = ttfont_filename(font)
+        names.append(file_name
+                     or ("<unknown %s>" % ttfont_format_as_extension(font)))
 
     return names
 
 
-def ttcfile_build(output_ttc_path, fontpath_list, tool_path=_BUILD_TOOL_PATH):
-    """Build a .ttc from a list of font files."""
-    otf2otc = tool_utils.resolve_path(tool_path)
-    if not otf2otc:
-        raise ValueError("can not resolve %s" % tool_path)
+def ttfont_filename(font):
+    font_fmt = ttfont_format_as_extension(font)
+    name_table = font.get("name")
+    if name_table:
+        ps_name = None
+        for r in name_table.names:
+            if (r.nameID, r.platformID, r.platEncID, r.langID) == (6, 3, 1, 0x409):
+                ps_name = r.string.decode("UTF-16BE")
+                break
```

Comment:
fonttools name table has a getDebugName() method that gives you an English name string, you can use that to get the PostScript name

Reply:
Thanks for the comment. The [logic](https://github.com/fonttools/fonttools/blob/main/Lib/fontTools/ttLib/tables/_n_a_m_e.py#L103) looks a little different, are we fine to change? This logic appears in the diff because I needed to move, but it is as it was in the original code.

Reply:
Updated the PR to use `getDebugName`.

Reply:
effectively it does the same thing, except that it attempts to get the Macintosh English name first (if any), but that's going to be exactly the same as the Windows English string so it should not matter.